providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/model/FlagParser.java

```diff
@@ -32,12 +33,13 @@ public class FlagParser {
     private static final String SCHEMA_RESOURCE = "flagd-definitions.json";
 
     private static final ObjectMapper MAPPER = new ObjectMapper();
-
     private static final Map<String, Pattern> PATTERN_MAP = new HashMap<>();
-    private static final Pattern REG_BRACKETS = Pattern.compile("^[^{]*\\{|}[^}]*$");
```

Comment:
Sonar didn't like the regex here. After considering the purpose of this, I attempted to avoid regex and use string replacements.